### PR TITLE
Add API

### PR DESCRIPTION
**Pull request by iross**
- Added base xDD API deployments/services
- Added prod credentials sealed secrets

@bbockelm can you look this over and tell me if:
- there's something obvious I misinterpreted
- There's extra cruft in the service/deployment that can be cleaned up. I based off what Jeff and I got successfully running (outside of gitops) last week.

### CODE REVIEW

File: manifests/base/xdd-api.yaml [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+    spec:
+      containers:
+      - image: iaross/xdd_api:redis
+        imagePullPolicy: Always
+        name: xdd-api
+        ports:
+        - containerPort: 5000
+          protocol: TCP
+        resources: {}
+        terminationMessagePath: /dev/termination-log
```
Comment (bbockelm) [inline manifests/base/xdd-api.yaml:32]: Just out of curiosity -- these seem to be a pretty detailed pod template.  Where'd you pick up these items?  Had to look up some of these keys.

File: manifests/base/xdd-api.yaml [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+apiVersion: v1
+kind: Service
+metadata:
+  name: xdd-api-webservice
+  labels:
+    app: xdd-api
+spec:
+  externalTrafficPolicy: Cluster
+  ports:
+  - nodePort: 31585
```
Comment (bbockelm) [inline manifests/base/xdd-api.yaml:57]: I don't think `nodePort` is needed here.  Please remove.

File: manifests/base/xdd-api.yaml [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+      role: master
+      tier: backend
+  strategy:
+    rollingUpdate:
+      maxSurge: 25%
+      maxUnavailable: 25%
+    type: RollingUpdate
+  template:
+    metadata:
+      creationTimestamp: null
```
Comment (bbockelm) [inline manifests/base/xdd-api.yaml:110]: I think this can be removed.
```suggestion
```

File: manifests/base/xdd-api.yaml [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+        ports:
+        - containerPort: 6379
+          protocol: TCP
+        resources:
+          requests:
+            cpu: 100m
+            memory: 100Mi
+        terminationMessagePath: /dev/termination-log
+        terminationMessagePolicy: File
+      dnsPolicy: ClusterFirst
```
Comment (bbockelm) [inline manifests/base/xdd-api.yaml:129]: This is already the default
```suggestion
```

File: manifests/base/xdd-api.yaml [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+        ports:
+        - containerPort: 5000
+          protocol: TCP
+        resources: {}
+        terminationMessagePath: /dev/termination-log
+        terminationMessagePolicy: File
+        volumeMounts:
+        - mountPath: /mnt/secrets
+          name: database-secrets-volume
+      dnsPolicy: ClusterFirst
```
Comment (bbockelm) [inline manifests/base/xdd-api.yaml:37]: This is already the default
```suggestion
```

File: manifests/base/xdd-api.yaml [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+          protocol: TCP
+        resources:
+          requests:
+            cpu: 100m
+            memory: 100Mi
+        terminationMessagePath: /dev/termination-log
+        terminationMessagePolicy: File
+      dnsPolicy: ClusterFirst
+      restartPolicy: Always
+      schedulerName: default-scheduler
```
Comment (bbockelm) [inline manifests/base/xdd-api.yaml:131]:
```suggestion
```

File: manifests/base/xdd-api.yaml [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+  - port: 6379
+    protocol: TCP
+    targetPort: 6379
+  selector:
+    app: redis
+    role: master
+    tier: backend
+  sessionAffinity: None
+  type: ClusterIP
+status:
```
Comment (bbockelm) [inline manifests/base/xdd-api.yaml:85]: Don't need to include the `status` stanza.

File: manifests/production/kustomization.yaml [hunk 8]
```diff
@@ -4,3 +4,5 @@ namespace: xdd
 bases:
   - ../base
 
+resources:
+  - ./xddsealedcredentials.yaml
```
Comment (bbockelm) [inline manifests/production/kustomization.yaml:8]:
```suggestion
  - xddsealedcredentials.yaml
```

File: manifests/production/xddsealedcredentials.yaml [hunk 9]
```diff
@@ -0,0 +1,15 @@
+apiVersion: bitnami.com/v1alpha1
+kind: SealedSecret
+metadata:
+  creationTimestamp: null
```
Comment (bbockelm) [inline manifests/production/xddsealedcredentials.yaml:4]:
```suggestion
```

File: manifests/production/xddsealedcredentials.yaml [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
+  template:
+    metadata:
+      creationTimestamp: null
```
Comment (bbockelm) [inline manifests/production/xddsealedcredentials.yaml:12]:
```suggestion
```

File: manifests/base/xdd-api.yaml (hunk 1 above)
Comment (iross) [reply manifests/base/xdd-api.yaml:32]: I can take zero credit. I did a dump of the running version that Jeff set up last week.

**Pull request merged**